packages/interactive_media_ads/lib/src/platform_interface/platform_ads_rendering_settings.dart

```diff
@@ -0,0 +1,134 @@
+
+import 'interactive_media_ads_platform.dart';
+import 'ui_element.dart';
+
+/// Object specifying creation parameters for creating a
+/// [PlatformAdsRenderingSettings].
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// This example demonstrates how to extend the
+/// [PlatformAdsRenderingSettingsCreationParams] to provide additional platform
+/// specific parameters.
+///
+/// When extending [PlatformAdsRenderingSettingsCreationParams], additional
+/// parameters should always accept `null` or have a default value to prevent
+/// breaking changes.
+///
+/// ```dart
+/// final class AndroidPlatformAdsRenderingSettingsCreationParams
+///     extends PlatformAdsRenderingSettingsCreationParams {
+///   const AndroidPlatformAdsRenderingSettingsCreationParams({
+///     super.bitrate,
+///     this.enableCustomTabs,
+///   }) : super();
+///
+///   factory AndroidPlatformAdsRenderingSettingsCreationParams.fromPlatformAdsRenderingSettingsCreationParams(
+///     PlatformAdsRenderingSettingsCreationParams params, {
+///     bool? enableCustomTabs,
+///   }) {
+///     return AndroidPlatformAdsRenderingSettingsCreationParams(
+///       bitrate: params.bitrate,
+///       enableCustomTabs: enableCustomTabs,
+///     );
+///   }
+///
+///   final bool? enableCustomTabs;
+/// }
+/// ```
+@immutable
+base class PlatformAdsRenderingSettingsCreationParams {
+  /// Used by the platform implementation to create a new
+  /// [PlatformAdsRenderingSettings].
+  const PlatformAdsRenderingSettingsCreationParams({
+    this.bitrate,
+    this.enablePreloading,
+    this.loadVideoTimeout,
+    this.mimeTypes,
+    this.playAdsAfterTime,
+    this.uiElements,
+  });
+
+  /// Maximum recommended bitrate.
+  ///
+  /// The value is in kbit/s.
+  ///
+  /// The SDK will select media which has a bitrate below the specified max or
+  /// the closest bitrate if there is no media with a lower bitrate found.
+  ///
+  /// If null, the bitrate will be selected by the SDK, using the currently
+  /// detected network speed (cellular or Wi-Fi).
+  final int? bitrate;
+
+  /// If set, the SDK will instruct the player to load the creative in response
+  /// initializing the ads manager.
+  ///
+  /// This allows the player to preload the ad at any point before starting the
+  /// ads manager.
+  ///
+  /// If null, the platform will decide the default value.
+  final bool? enablePreloading;
+
+  /// Specifies a non-default amount of time to wait for media to load before
+  /// timing out, in milliseconds.
+  ///
+  /// This only applies to the IMA client-side SDK. Default time is 8000 ms.
+  final int? loadVideoTimeout;
+
+  /// The SDK will prioritize the media with MIME type on the list.
+  ///
+  /// If empty, the SDK will pick the media based on player capabilities. This
+  /// only refers to the mime types of videos to be selected for linear ads.
+  final List<String>? mimeTypes;
```

Comment:
Non-nullable?

Reply:
Only the documentation for iOS actually states what happens if it is empty. Android doesn't actually mention it:

```
If specified, the SDK will prioritize the media with MIME type on the list. If not specified,
the SDK will pick the media based on player capabilities. This only refers to the mime
types of videos to be selected for linear ads.
```

This is related to having a "don't set" option for optional native parameters.

Reply:
That's an implementation detail though; if we want empty to mean "the SDK decides" as the docs currently say, then we should map an empty list to not setting on platforms that treat null as not setting.

If we actually want to treat `[]` and `null` differently, then we need to explain what the difference is in the comments, ideally in a cross-platform way.